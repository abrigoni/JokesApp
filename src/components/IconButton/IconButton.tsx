import React from 'react';
import {Pressable} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import {IconProps} from 'react-native-vector-icons/Icon';


const IconButton = ({onPress, ...props}: IconProps) => {
  return (
    <Pressable onPress={onPress}>
      <Icon {...props} />
    </Pressable>
  );
};

export default IconButton;
