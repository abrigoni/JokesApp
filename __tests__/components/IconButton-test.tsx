import 'react-native';
import React from 'react';
import IconButton from '../../src/components/IconButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');

describe('<IconButton />', () => {
  it('renders correctly', () => {
    const json = renderer
      .create(<IconButton iconProps={{name: 'Trash'}} />)
      .toJSON();
    expect(json).toMatchSnapshot();
  });
});
