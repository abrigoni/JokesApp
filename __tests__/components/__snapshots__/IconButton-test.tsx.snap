// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Icon
    name="Trash"
  />
</View>
`;

exports[`<IconButton /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Icon
    name="Trash"
  />
</View>
`;
